packages/medusa/src/api-v2/admin/products/route.ts

```diff
@@ -1,22 +1,58 @@
+import { createProductsWorkflow } from "@medusajs/core-flows"
+import { CreateProductDTO } from "@medusajs/types"
+import {
+  ContainerRegistrationKeys,
+  isString,
+  remoteQueryObjectFromString,
+} from "@medusajs/utils"
 import {
   AuthenticatedMedusaRequest,
   MedusaResponse,
 } from "../../../types/routing"
-
-import { CreateProductDTO } from "@medusajs/types"
-import { createProductsWorkflow } from "@medusajs/core-flows"
-import { remoteQueryObjectFromString } from "@medusajs/utils"
+import { listPriceLists } from "../price-lists/queries"
 
 export const GET = async (
   req: AuthenticatedMedusaRequest,
   res: MedusaResponse
 ) => {
-  const remoteQuery = req.scope.resolve("remoteQuery")
+  const remoteQuery = req.scope.resolve(ContainerRegistrationKeys.REMOTE_QUERY)
+  const filterableFields = { ...req.filterableFields }
+  const filterByPriceListIds =
+    (filterableFields.price_list_id as string[]) || []
```

Comment:
I think this can get messy if for the filter we have something like `$and: [{price_list_id: [1]}, {...}]` (this will be a general problem if we want to operate on the filters in any way).  Let's add a TODO at the very least, and maybe start a discussion in eng-core

Reply:
from an API point of view we do not allow such object shape with $and or $or, I believe it should be fine as it is defined by the validator

Reply:
I've linked the `filterableFields` type to the validator type and removed the cast. If anything in the API validators change in the future, this here will explode.